pkg/apis/apps/v1alpha1/apicast_types.go

```diff
@@ -16,15 +16,17 @@ type APIcastSpec struct {
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
 	// +optional
-	Replicas                  *int64                   `json:"replicas,omitempty"`
-	AdminPortalCredentialsRef *v1.LocalObjectReference `json:"adminPortalCredentialsRef"`
+	Replicas *int64 `json:"replicas,omitempty"`
+	// +optional
+	AdminPortalCredentialsRef *v1.LocalObjectReference `json:"adminPortalCredentialsRef,omitempty"`
```

Comment:
I have changed AdminPortalCredentialsRef to be optional instead of mandatory, because now you can deploy apicast specifying the AdminPortal credentials or the Apicast configuration in json in itself (via another secret).

However, this has a downside:
By making both elements optional the user might wrongly deploy an Apicast without setting any of them.
I've added code so the operator returns an error when this happens but it's not ideal.
What do you think about this?

Reply:
and let apicast do that?

Reply:
As we talked, for the moment we'll check it in the operator so it "breaks" before.
We'll see whether we could control this with OpenAPI in the future. It might require restructuration of the fields though

Reply:
Thinking more about it I removed the check in the operator, so we do not tie this behaviour into it in case we want to change it in the future with OpenAPI.